rpc/src/poster.rs

```diff
@@ -0,0 +1,176 @@
+                                        }
+                                    }
+                                    Err(e) => {
+                                        debug!("Error receiving response: {}", e);
+                                    }
+                                }
+                            }
+                            _ => {
+                                let result = connection.clean_response(total_length);
+                                match result {
+                                    Ok(_) => {}
+                                    Err(e) => {
+                                        debug!("Error cleaning up response: {}", e);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    Err(e) => {
+                        debug!("Error parsing header: {}", e);
+                    }
+                }
+            }
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn call_remote(
+        &self,
+        connection_index: i32,
+        operation_type: u32,
+        req_flags: u32,
+        path: &str,
+        send_meta_data: &[u8],
+        send_data: &[u8],
+        status: &mut i32,
+        rsp_flags: &mut u32,
+        recv_meta_data: &mut [u8],
+        recv_data: &mut [u8],
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        debug!("call_remote on connection: {}", connection_index);
+        let connection = self.connections.get(&connection_index).unwrap();
+        debug!(
+            "call_remote on connection: {:?}",
+            connection.value().server_address
+        );
+        let result = self.queue.register_callback(recv_meta_data, recv_data);
+        debug!("call_remote on connection: {:?}", result);
+        match result {
+            Ok(id) => {
+                let send_result = connection.send_request(
+                    id,
+                    operation_type,
+                    req_flags,
+                    path,
+                    send_meta_data,
+                    send_data,
+                );
+                match send_result {
+                    Ok(_) => {
+                        let result = self.queue.wait_for_callback(id);
+                        match result {
+                            Ok(_) => {
+                                *status = self.queue.get_status(id).unwrap();
+                                *rsp_flags = self.queue.get_rsp_flags(id).unwrap();
+                                Ok(())
+                            }
+                            Err(_) => Err(Box::new(std::io::Error::new(
+                                std::io::ErrorKind::Other,
+                                "Error waiting for callback",
+                            ))),
+                        }
+                    }
+                    Err(_) => Err(Box::new(std::io::Error::new(
```

Comment:
Do `CircularQueue` need to call `clean_up`?

Reply:
Yes, I've done this.